packages/ckeditor5-coremedia-blocklist/src/blocklistediting.ts

```diff
@@ -0,0 +1,319 @@
+      .catch((reason) => {
+        BlocklistEditing.#logger.warn("Error while adding word to blocklist", reason);
+      });
+  }
+
+  /**
+   * Removes a word from the blocklist.
+   * Use this method for changes, triggered by the UI, not by the service.
+   *
+   * This method updates the internal list, changes the markers accordingly
+   * and triggers the blocklistService.
+   *
+   * @param wordToUnblock - the word to remove from the list
+   */
+  removeBlocklistWord(wordToUnblock: string): void {
+    // Update internal list
+    this.#removeFromInternalBlocklist(wordToUnblock);
+    // Update markers
+    this.#removeMarkersForWord(wordToUnblock);
+    // Update value in service
+    serviceAgent
+      .fetchService(createBlocklistServiceDescriptor())
+      .then((blocklistService: BlocklistService) => blocklistService.removeFromBlocklist(wordToUnblock))
+      .catch((reason) => {
+        BlocklistEditing.#logger.warn("Error while removing word from blocklist", reason);
+      });
+  }
+
+  /**
+   * Adds a word to the internal blocklist.
+   * Words will be transformed to lowercase before added.
+   *
+   * @param wordToBlock - the word to add
+   * @private
+   */
+  #addToInternalBlocklist(wordToBlock: string) {
+    const lowerCaseWord = wordToBlock.toLowerCase();
+    if (!this.internalBlocklist.includes(lowerCaseWord)) {
+      this.internalBlocklist.push(lowerCaseWord);
+    }
+  }
+
+  /**
+   * Removes a word from the internal blocklist.
+   *
+   * @param wordToUnblock - the word to remove
+   * @private
+   */
+  #removeFromInternalBlocklist(wordToUnblock: string) {
+    const lowerCaseWord = wordToUnblock.toLowerCase();
+    if (!this.internalBlocklist.includes(lowerCaseWord)) {
+      this.internalBlocklist = this.internalBlocklist.filter((word) => word !== lowerCaseWord);
+    }
+  }
+
+  /**
+   * Removes all markers for a given blocklisted word.
+   * This means:
+   * * The local collection of markers will be cleared of
+   * all markers for the given word.
+   * * The model is adjusted accordingly.
+   *
+   * @param wordToUnblock - the word, defining the markers that will be removed
+   */
+  #removeMarkersForWord(wordToUnblock: string): void {
+    const model = this.editor.model;
+    const markers = this.blockedWordMarkers;
+    const markersToRemove: string[] = [];
+    model.change((writer) => {
+      markers.map((resultType) => {
+        const markerName = resultType.marker?.name;
+        if (markerName === undefined || resultType.id === undefined) {
+          return;
+        }
+        const { blockedWord } = this.#splitMarkerName(markerName);
+        if (wordToUnblock === blockedWord) {
+          writer.removeMarker(markerName);
+          markersToRemove.push(resultType.id);
+        }
+      });
+    });
+    markersToRemove.forEach((markerId) => {
+      markers.remove(markerId);
+    });
+  }
+
+  /**
+   * Adds markers for a given blocklisted word.
```

Comment:
Just a wording thing which may be discussed: blocklisted word seems weird, isn't it just blocked word?